app/src/main/java/crux/bphc/cms/fragments/CourseSectionFragment.java

```diff
@@ -318,46 +315,67 @@ private String addToken(String descriptionWithOutExtraSpace) {
 
     public void makeTextViewResizable(final TextView description, final int maxLine, final String expandText, final boolean viewMore) {
 
-        if (description.getTag() == null) {
-            description.setTag(description.getText());
-        }
         ViewTreeObserver vto = description.getViewTreeObserver();
         vto.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
 
 
             @Override
             public void onGlobalLayout() {
-                String text;
-                int lineEndIndex;
+                String displayText; /* The final text that is to be displayed  */
+                String descriptionHtml = (String) description.getTag(); /* The description of the course section (with HTML tags) */
+                String descriptionText = Html.fromHtml(descriptionHtml).toString(); /* The description of the course section (with the HTML tags removed) */
+
                 ViewTreeObserver obs = description.getViewTreeObserver();
                 obs.removeOnGlobalLayoutListener(this);
-                if (maxLine == 0) {
-                    text = expandText;
-                } else if (maxLine > 0 && description.getLineCount() > maxLine) {
-                    lineEndIndex = description.getLayout().getLineEnd(maxLine - 1);
-                    text = description.getText().subSequence(0, lineEndIndex) + "\n" + expandText;
-                } else if (description.getLineCount() <= maxLine) {
-                    text = description.getText().toString();
+                if (viewMore) {
+                    if (maxLine == 0) {
+                        displayText = expandText;
+                    } else if (description.getLineCount() > maxLine) {
+                        int lineEndIndex = description.getLayout().getLineEnd(maxLine - 1); /* The maximum number of characters that the TextView can show at a time */
+                        /* Find out how much is actually visible (without HTML tags) and grab the HTML description till that part */
+                        if (descriptionText.length() > lineEndIndex) {
+
+                            /* Start from lineEndIndex and find out the next complete word */
+                            String nearestWord = Util.nextNearestWord(descriptionText, lineEndIndex);
+                            displayText = descriptionHtml.subSequence(0, descriptionHtml.indexOf(nearestWord)).toString() + "\n" + expandText;
+
+                        }else displayText = descriptionHtml;
+                    } else if (description.getLineCount() <= maxLine) {
+                        displayText = descriptionText;
+                    } else {
```

Comment:
Under what conditions is this `else` block triggered? Aren't the previous 3 conditions exhaustive?